iroha/src/kura.rs

```diff
@@ -180,77 +180,129 @@ impl Default for Mode {
 #[derive(Debug)]
 pub struct BlockStore {
     path: PathBuf,
+    blocks_per_file: u64,
 }
 
 impl BlockStore {
     /// Initialize block storage at `path`.
     ///
     /// # Errors
     /// - Failed to create directory.
-    pub fn new(path: &Path) -> Result<BlockStore> {
+    pub fn new(path: &Path, blocks_per_file: u64) -> Result<BlockStore> {
         if fs::read_dir(path).is_err() {
             fs::create_dir_all(path).wrap_err("Failed to create Block Store directory.")?;
         }
         Ok(BlockStore {
             path: path.to_path_buf(),
+            blocks_per_file: if blocks_per_file == 0 { config::DEFAULT_BLOCKS_PER_STORAGE_FILE } else { blocks_per_file },
```

Comment:
It is bad api in rust to encode one of integer variants as special thing. Enums suit better for that.

Reply:
NonZero as well then?